ebpfsvc/svcmain.cpp

```diff
@@ -12,23 +12,23 @@ SERVICE_STATUS_HANDLE ebpf_service_status_handle;
 HANDLE ebpf_service_stop_event_handle = nullptr;
 
 void WINAPI
-service_control_handler(DWORD ctrl);
+service_control_handler(unsigned long ctrl);
 void
-service_report_event(PTSTR function);
+service_report_event(_Null_terminated_ wchar_t* function);
 void
-report_service_status(DWORD current_state, DWORD win32exitcode, DWORD wait_hint);
+report_service_status(unsigned long current_state, unsigned long win32exitcode, unsigned long wait_hint);
 void
-service_init(DWORD argc, PTSTR* argv);
+service_init(unsigned long argc, _Null_terminated_ wchar_t** argv);
 
 void WINAPI
-service_main(DWORD argc, PTSTR* argv);
+service_main(unsigned long argc, _Null_terminated_ wchar_t** argv);
 int
 service_install();
 
-int __cdecl wmain(ULONG argc, PWSTR* argv)
+int __cdecl wmain(unsigned long argc, wchar_t** argv)
 {
     SERVICE_TABLE_ENTRY dispatch_table[] = {
-        {(PWSTR)SERVICE_NAME, (LPSERVICE_MAIN_FUNCTION)service_main}, {nullptr, nullptr}};
+        {(wchar_t*)SERVICE_NAME, (LPSERVICE_MAIN_FUNCTION)service_main}, {nullptr, nullptr}};
```

Comment:
Can we get rid of the LP part of LPSERVICE_MAIN_FUNCTION and use SERVICE_MAIN_FUNCTION* ?

Reply:
+1